conjure-java/src/main/java/com/palantir/conjure/java/cli/ConjureJavaCli.java

```diff
@@ -127,31 +133,31 @@ public void run() {
         @SuppressWarnings("BanSystemErr")
         public void run() {
             CliConfiguration config = getConfiguration();
-            if (config.generateObjects() && !config.featureFlags().contains(FeatureFlags.UseImmutableBytes)) {
-                System.err.println("[WARNING] Using deprecated ByteBuffer codegen, please enable the "
-                        + "--useImmutableBytes feature flag to opt into the preferred implementation");
-            }
             try {
                 ConjureDefinition conjureDefinition = OBJECT_MAPPER.readValue(config.input(), ConjureDefinition.class);
-                TypeGenerator typeGenerator = new ObjectGenerator(config.featureFlags());
-                ServiceGenerator jerseyGenerator = new JerseyServiceGenerator(config.featureFlags());
-                ServiceGenerator retrofitGenerator = new Retrofit2ServiceGenerator(config.featureFlags());
-                ServiceGenerator undertowGenerator = new UndertowServiceGenerator(config.featureFlags());
-
                 if (config.generateObjects()) {
-                    typeGenerator.emit(conjureDefinition, config.outputDirectory());
+                    if (!config.featureFlags().contains(FeatureFlags.UseImmutableBytes)) {
+                        System.err.println("[WARNING] Using deprecated ByteBuffer codegen, please enable the "
+                                + "--useImmutableBytes feature flag to opt into the preferred implementation");
+                    }
+                    new ObjectGenerator(config.featureFlags()).emit(conjureDefinition, config.outputDirectory());
+                }
+                if (config.generateJaxRsClient()) {
+                    new JaxRsClientGenerator(config.featureFlags()).emit(conjureDefinition, config.outputDirectory());
                 }
                 if (config.generateJersey()) {
```

Comment:
thoughts about adding ServiceGenerator to a list based on the flag and then `list.foreach(ServiceGenerator::emit)` ?

Reply:
I think there's something clever here using a `BiConsumer<Definition,Output>` and chaining via `andThen(ServiceGenerator::emit)`, but I think the code is easier to follow as is.